src/dask_awkward/lib/core.py

```diff
@@ -398,6 +399,10 @@ def name(self) -> str:
     def key(self) -> Key:
         return (self._name, 0)
 
+    @property
+    def report(self):
+        return getattr(self._meta, "_report", set())
```

Comment:
Can we lift `_report` from the meta to the Layer? I know that it's more state to keep in sync, but I don't love the act of tagging non-reserved attributes onto an Awkward Array, and this will better separate the stateful part of typetracer from the stateless part.

Reply:
@jpivarski I wonder whether we should add a new `merge` function to `TypeTracerReport`. Due to the uniqueness of layer IDs, and the existing association of reports with layouts, we can replace the need to carry a set of reports with a single report via
```
merge(A, B) -> A
```

I haven't yet thought about the memory implication for effectively having "larger" reports, but this might be a nicer UX.